src/infrastructure/NodeService.js

```diff
@@ -195,6 +195,10 @@ class NodeService {
             formattedNode.rolesRaw === 6 ||
             formattedNode.rolesRaw === 7
     	) {
+    		// If https not working, it will replace to http with port 3000
+    		if (!(await this.isHttpsEnabled(formattedNode.apiEndpoint)))
+    			formattedNode.apiEndpoint = formattedNode.apiEndpoint.replace('https', 'http').replace(':3001', ':3000');
```

Comment:
what about calling the HTTPS URL by default, if the exception is thrown, try HTTP, rather than always makes an extra call?

Reply:
If you are trying a URL, remember to add a timeout. If the port is close, the request may hang

Reply:
Yes, I make a timeout `3000` in Axios call. it's good enough?